controllers/control_plane_init.go

```diff
@@ -0,0 +1,48 @@
+package controllers
+
+import (
+	"encoding/json"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/api/v1alpha1"
+	"sigs.k8s.io/cluster-api-bootstrap-provider-kubeadm/cloudinit"
+)
+
+type UserDataGenerator struct {
+	logr.Logger
+}
+
+func (u *UserDataGenerator) GenerateControlPlaneInit(config *v1alpha1.KubeadmBootstrapConfig) error {
+	u.Info("Generating user data for first control plane")
+
+	// TODO: sub in things in the cluster configuration like the cluster name and other data we can extract from the cluster object
+	config.Spec.ClusterConfiguration.APIVersion = "v1beta1"
+	config.Spec.ClusterConfiguration.Kind = "ClusterConfiguration"
+	clusterdata, err := json.Marshal(config.Spec.ClusterConfiguration)
```

Comment:
I'm a little bit confused about this because `GenerateControlPlaneInit` in part is setting/enforcing values in the config, and in other parts it is consuming the config (like here)
What about moving all the consume config part into `cloudinit.NewInitControlPlane`?

Reply:
That's fine, I hadn't spent time structuring this code. I plan to improve this as we add more than just control-plane-init generation and if that refactor makes sense I will use it.